src/int.rs

```diff
@@ -0,0 +1,1409 @@
+    pub fn saturating_mul(self, rhs: Self) -> Self {
+        self.checked_mul(rhs).unwrap_or_else(|| {
+            match Sign::from_signum64(self.signum64() * rhs.signum64()) {
+                Sign::Positive => I256::MAX,
+                Sign::Negative => I256::MIN,
+            }
+        })
+    }
+
+    /// Wrapping multiplication.
+    pub fn wrapping_mul(self, rhs: Self) -> Self {
+        let (result, _) = self.overflowing_mul(rhs);
+        result
+    }
+
+    /// Returns the sign of `self` to the exponent `exp`.
+    ///
+    /// Note that this method does not actually try to compute the `self` to the
+    /// exponent `exp`, but instead uses the property that a negative number to
+    /// an odd exponent will be negative. This means that the sign of the result
+    /// of exponentiation can be computed even if the actual result is too large
+    /// to fit in 256-bit signed integer.
+    fn pow_sign(self, exp: u32) -> Sign {
+        let is_exp_odd = exp % 2 != 0;
+        if is_exp_odd && self.is_negative() {
+            Sign::Negative
+        } else {
+            Sign::Positive
+        }
+    }
+
+    /// Create 10**n as this type.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the result overflows the type.
+    pub fn exp10(n: usize) -> Self {
+        U256::exp10(n).try_into().expect("overflow")
+    }
+
+    /// Raise self to the power of `exp`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the result overflows the type in debug mode.
+    pub fn pow(self, exp: u32) -> Self {
+        handle_overflow(self.overflowing_pow(exp))
+    }
+
+    /// Raises self to the power of `exp`.
+    ///
+    /// Returns a tuple of the exponentiation along with a bool indicating
+    /// whether an overflow happened.
+    pub fn overflowing_pow(self, exp: u32) -> (Self, bool) {
+        let sign = self.pow_sign(exp);
+        let (unsigned, overflow_pow) = self.abs_unsigned().overflowing_pow(exp.into());
+        let (result, overflow_conv) = I256::overflowing_from_sign_and_abs(sign, unsigned);
+
+        (result, overflow_pow || overflow_conv)
+    }
+
+    /// Raises self to the power of `exp`. Returns None if overflow occurred.
+    pub fn checked_pow(self, exp: u32) -> Option<Self> {
+        let (result, overflow) = self.overflowing_pow(exp);
+        if overflow {
+            None
+        } else {
+            Some(result)
+        }
+    }
+
+    /// Raises self to the power of `exp`, saturating at the numeric bounds
+    /// instead of overflowing.
+    pub fn saturating_pow(self, exp: u32) -> Self {
+        let (result, overflow) = self.overflowing_pow(exp);
+        if overflow {
+            match self.pow_sign(exp) {
+                Sign::Positive => I256::MAX,
+                Sign::Negative => I256::MIN,
+            }
+        } else {
+            result
+        }
+    }
+
+    /// Wrapping powolute value. Computes `self.pow()`, wrapping around at the
```

Comment:
What is "powolute"? Is this some combination of power and absolute? Absolute Power. If I remember correctly, this was Jafar's last wish that turned him into a genie and saved the day for everyone else.

Reply:
:man_facepalming: That is yet another typo.

Reply:
I had done a search and replace all of `abs` for `pow` and this comment got caught in the crossfire :stuck_out_tongue_closed_eyes: